core/src/test/java/com/linecorp/armeria/server/file/FileServiceTest.java

```diff
@@ -297,13 +297,20 @@ void testAutoIndex(String baseUri) throws Exception {
         final String basePath = new URI(baseUri).getPath();
 
         try (CloseableHttpClient hc = HttpClients.createMinimal()) {
-            // Ensure auto-redirect works as expected.
+            // Ensure auto-redirect without query works as expected.
             HttpUriRequest req = new HttpGet(baseUri + "/fs/auto_index");
             try (CloseableHttpResponse res = hc.execute(req)) {
                 assertStatusLine(res, "HTTP/1.1 307 Temporary Redirect");
                 assertThat(header(res, "location")).isEqualTo(basePath + "/fs/auto_index/");
             }
 
+            // Ensure auto-redirect with query works as expected.
+            req = new HttpGet(baseUri + "/fs/auto_index?foobar=1");
+            try (CloseableHttpResponse res = hc.execute(req)) {
+                assertStatusLine(res, "HTTP/1.1 307 Temporary Redirect");
+                assertThat(header(res, "location")).isEqualTo(basePath + "/fs/auto_index/?foobar=1");
```

Comment:
optional; would it also make sense to add a test which ensures directory listing works as expected with query params?

Reply:
I've added some more test cases per your suggestion